modelforge/tests/test_curation.py

```diff
@@ -1450,216 +1481,490 @@ def test_spice114_process_download_conversion(prep_temp_dir):
-#         )
-#     )
-#     # spot check energy
-#     assert (
-#         spice_openff_data.data[0]["dft_total_energy"][0][0]
-#         == -1168.2328704724725 * unit.hartree
-#     )
-#     assert (
-#         spice_openff_data.data[4]["dft_total_energy"][0][0]
-#         == -2011.874682644447 * unit.hartree
-#     )
-
-#     # check the shape of a molecule with only a single conformer in the test set
-#     assert spice_openff_data.data[0]["atomic_numbers"].shape == (47, 1)
-#     assert spice_openff_data.data[0]["geometry"].shape == (1, 47, 3)
-#     assert spice_openff_data.data[0]["dft_total_energy"].shape == (1, 1)
-#     assert spice_openff_data.data[0]["dft_total_gradient"].shape == (1, 47, 3)
-#     assert spice_openff_data.data[0]["mbis_charges"].shape == (1, 47, 1)
-#     assert spice_openff_data.data[0]["scf_dipole"].shape == (1, 3)
-#     assert spice_openff_data.data[0]["formation_energy"].shape == (1, 1)
-#     assert spice_openff_data.data[0]["formation_energy"].shape == (1, 1)
-
-#     # check the shape of a molecule with multiple conformers in the test set
-#     assert spice_openff_data.data[4]["atomic_numbers"].shape == (16, 1)
-#     assert spice_openff_data.data[4]["geometry"].shape == (6, 16, 3)
-#     assert spice_openff_data.data[4]["dft_total_energy"].shape == (6, 1)
-#     assert spice_openff_data.data[4]["dft_total_gradient"].shape == (6, 16, 3)
-#     assert spice_openff_data.data[4]["mbis_charges"].shape == (6, 16, 1)
-#     assert spice_openff_data.data[4]["scf_dipole"].shape == (6, 3)
-#     assert spice_openff_data.data[4]["formation_energy"].shape == (6, 1)
+def test_ani2x(prep_temp_dir):
+    from modelforge.curation.ani2x_curation import ANI2xCuration
+
+    local_path_dir = str(prep_temp_dir)
+    local_data_path = resources.files("modelforge").joinpath("tests", "data")
+
+    # create an hdf5 file that only contains the dimer data, generated using the following script
+    # so we can test the _process_downloaded function
+    # with h5py.File("ANI-2x-wB97X-631Gd.h5", "r") as hf_in:
+    #     with h5py.File('ani2x_n10.hdf5', "w") as hf_out:
+    #         for num_atoms, properties in hf_in.items():
+    #             hf_in.copy(hf_in[num_atoms], hf_out)
+    #             break
+
+    filename = "ani2x_minimal.hdf5"
+
+    ani2x_dataset = ANI2xCuration(
+        hdf5_file_name="ani2x_dataset.hdf5",
+        output_file_dir=local_path_dir,
+        local_cache_dir=local_path_dir,
+    )
+    ani2x_dataset._process_downloaded(
+        local_data_path, filename, unit_testing_max_records=1
+    )
+
+    assert len(ani2x_dataset.data) == 1
+    assert ani2x_dataset.data[0]["name"] == "[1_9]"
+    assert ani2x_dataset.data[0]["n_configs"] == 517
+    assert ani2x_dataset.data[0]["energies"].shape == (517, 1)
+    assert ani2x_dataset.data[0]["atomic_numbers"].shape == (2, 1)
+    assert ani2x_dataset.data[0]["geometry"].shape == (517, 2, 3)
+    assert ani2x_dataset.data[0]["forces"].shape == (517, 2, 3)
```

Comment:
Maybe I misremember this, but I thought that the ANI2x dataset contains forces with two levels of theory.

Reply:
Ani1x has different levels of theory, in the same source hdf5 file.  Ani2x splits them up into separate files.  This is the data from the main paper (and what  is used in the published model).  I need to look at each of the files a bit more closely.  they might be completely interchangeable in terms of the curation code (so we can just add in a flag to set the level of theory).  I think the class for loading in this data would be the same sort of idea, which file it will use, but we need to be better at naming conventions of temporary files.

Reply:
Talked to John about this today;  we are going to just stick with this dataset to start.